File: frontend/Operations_Manager/src/app/service/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User } from '../models/user';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  private apiUrl = 'http://localhost:8081/api/users';

  constructor(private http: HttpClient) {}

  getUsers(): Observable<User[]>
  {

    return this.http.get<User[]>(`${this.apiUrl}/getAllUsers`);
  }

  getUser(id: string): Observable<User>
  {
    return this.http.get<User>(`${this.apiUrl}/${id}`);
  }

  addUser(user: User): Observable<User>
  {
    return this.http.post<User>(this.apiUrl, user);
  }

  updateUser(user: User): Observable<User>
  {
    return this.http.put<User>(`${this.apiUrl}/${user.id}`, user);
  }

  deleteUser(id: string): Observable<void>
  {
    return this.http.delete<void>(`${this.apiUrl}/${id}`); }
  }

File: backend/operations/src/main/java/com/vcapone/operations/service/UserService.java
package com.vcapone.operations.service;

import java.util.List;
import java.util.UUID;

import com.vcapone.operations.entity.User;

public interface UserService {
    List<User> getAllUsers(); 
    User getUserById(UUID id); 
    User addUser(User user); 
    User updateUser(UUID id, User user); 
    void deleteUser(UUID id);
}
